scripts/generate-docs-types.js

```diff
@@ -0,0 +1,287 @@
+const typeRegex = /:\s(\w+);$/i;
+
+// const signature = (type, args, returnType) => {
+const signature = (type, args) => {
+  const required = args
+    .map(line => {
+      const match = line.match(callRegex);
+      if (!match || match[2] === "?:") {
+        return;
+      }
+      return match[1].slice(0, 1) === "_" ? match[1].slice(1) : match[1];
+    })
+    .filter(s => Boolean(s))
+    .join(", ");
+  return [
+    "```javascript",
+    `t.${type}(${required})`,
+    // `t.${type}(${required}): ${returnType || "unknown"}`,
+    "```",
+    "",
+  ];
+};
+
+const removeNullUndefined = type => {
+  if (type.includes("null") || type.includes("undefined")) {
+    const types = type.split(" | ");
+    return types.filter(a => a !== "null" && a !== "undefined").join(" | ");
+  }
+  return type;
+};
+
+const typeOverwrites = {
+  classMethod: {
+    key: "if computed then `Expression` else `Identifier | Literal`",
+  },
+  memberExpression: {
+    property: "if computed then `Expression` else `Identifier`",
+  },
+  objectMethod: {
+    key: "if computed then `Expression` else `Identifier | Literal`",
+  },
+  objectProperty: {
+    key: "if computed then `Expression` else `Identifier | Literal`",
+  },
+};
+
+const requirementOverwrites = {
+  classMethod: {
+    kind: "default: `method`",
+  },
+  program: {
+    sourceType: "default: `'script'`",
+  },
+  regExpLiteral: {
+    flags: 'default: `""`',
+  },
+  tsDeclareMethod: {
+    kind: "default: `'method'`",
+  },
+};
+
+const getRequirement = (required, id, type, types) => {
+  if (
+    requirementOverwrites.hasOwnProperty(type) &&
+    requirementOverwrites[type].hasOwnProperty(id)
+  ) {
+    return requirementOverwrites[type][id];
+  }
+
+  if (required) {
+    return "required";
+  }
+
+  if (types.slice(0, "`Array<".length) === "`Array<") {
+    return "default: `[]`";
+  }
+
+  switch (types) {
+    case "`boolean`":
+      return "default: `false`";
+
+    case "`string`":
+      return 'default: `""`';
+
+    default:
+      return "default: `null`";
+  }
+};
+
+const typeToDoc = type => line => {
+  const match = line.match(callRegex);
+  const id = match[1].slice(0, 1) === "_" ? match[1].slice(1) : match[1];
+  const required = match[2] === ":";
+  const types =
+    typeOverwrites.hasOwnProperty(type) &&
+    typeOverwrites[type].hasOwnProperty(id)
+      ? typeOverwrites[type][id]
+      : "`" + removeNullUndefined(match[3]) + "`";
+  return ` - \`${id}\`: ${types} (${getRequirement(
+    required,
+    id,
+    type,
+    types
+  )})`;
+};
+
+const seeAlso = type => {
+  const see = { assert: "", is: "" };
+  const is = String(`is${type}`).toLowerCase();
+  const assert = String(`assert${type}`).toLocaleLowerCase();
+  for (const type in t) {
+    if (type.toLowerCase() === assert) {
+      see.assert = type;
+    }
+    if (type.toLowerCase() === is) {
+      see.is = type;
+    }
+  }
+  return `See also \`t.${see.is}(node, opts)\` and \`t.${
```

Comment:
what if `see.is` or `see.assert` is not available, wouldn't it be `See also and ` ?
maybe a join with `" and "` would be better?

Reply:
and if we adding `is` and `assert` section, maybe this can be a link to the section